### ISSUE TITLE
Parameterize/inject config from gradle

### ISSUE THREAD

**Issue by aknauf** [detekt/detekt#4936]
## Expected Behavior
I want to apply the same `detekt.yml` config file to all subprojects in my project, but update/override the parameters to the applied rules, on a per-module basis.

## Current Behavior
Currently, there seems to be no way to express in the config file that I want each subproject to have basically the same ruleset, but with some sub-project-specific parameters - such as `rootPackage`.

My options seem to be to either disable the new `InvalidPackageDeclaration` rule, introduce with 1.21-RC1, or to to separately apply the plugin to each sub-project, copy-pasting the entire config file, just to change that one parameter. Of course, I can write custom code to merge/template my detekt `config.yml`, but it seems to me that a way to inject parameters by subproject would broadly useful.

Perhaps, something like:
```kotlin
tasks.withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
  detekt.rule("InvalidPackageDeclaration").parameter("rootPackage").set("$group.${project.name}")
}
```

## Context
In my project, we follow the kotlin coding conventions and our "directory structure follows the package structure with the common root package omitted". Each sub-project in the project has a different common root pacakage. When applying the new `InvalidPackageDeclaration` rule, I want to supply a different `rootPackage` for each sub-project.

## Further considerations
When using the IntelliJ detekt plugin, I would want to be sure that IntelliJ could also recognise the injected rule parameters.

Am I missing a simpler fix here?

**Comment by aknauf** [comment]
Ok, so I've realised that there is indeed a simpler fix.  I can have a config file in the root directory and then override it at the submodule level with a minimal config file. e.g. 

 - `a/build.gradle.kts`
```kotlin
detekt {
    config = files("../config/detekt/detekt.yml", "config/detekt/detekt.yml")
}
```

 - `/config/detekt/detekt.yml`
```yaml
<snip>
naming:
  active: true
<snip>
  FunctionParameterNaming:
    active: true
    parameterPattern: '[a-z][A-Za-z0-9]*'
    excludeClassPattern: '$^'
    ignoreOverridden: true
  InvalidPackageDeclaration:
    active: true
#    rootPackage: ''
    requireRootInDeclaration: true
  LambdaParameterNaming:
    active: false
    parameterPattern: '[a-z][A-Za-z0-9]*|_'
<snip>
```
 - `a/config/detekt/detekt.yml`
```yaml
naming:
  InvalidPackageDeclaration:
    rootPackage: 'com.example.a'
```

 - `b/config/detekt/detekt.yml`
```yaml
naming:
  InvalidPackageDeclaration:
    rootPackage: 'com.example.b'
```

This solves the issue for me, so feel free to disregard. :-)

**Comment by amal** [comment]
Not a nice thing to require Detekt config in each of the potentially many modules.

**Comment by BraisGabin** [comment]
Gradle gives you plenty of options to don't repeat that configuration. The easiest: 

``` kotlin
subpojects {
  detekt {
    // Your magic here
  }
}
```
